app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java

```diff
@@ -57,19 +56,62 @@ public void dataReceived(final byte[] bytes) {
      * {@inheritDoc}
      */
     @Override
-    public void send(final Serializable object) throws IOException {
+    public final void send(final Serializable object) {
         LOGGER.info("Begin writing object");
-        final byte[] bytes;
+        final byte[] bytes = objectToBytes(object);
+        this.send(bytes);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public final void send(final Serializable object, final int key) {
+        LOGGER.info("Begin writing object with encryption");
+        final byte[] bytes = objectToBytes(object);
+        // TODO encrypt the bytes
+        this.send(bytes);
+    }
+
+    protected final IServiceLocator getServiceLocator() {
+        return this.serviceLocator;
+    }
+
+    private static byte[] objectToBytes(final Serializable object) {
         try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
              ObjectOutputStream oos = new ObjectOutputStream(bos)) {
             oos.writeObject(new DataWrapper(object));
             oos.flush();
-            bytes = bos.toByteArray();
+            return bos.toByteArray();
+        } catch (final IOException e) {
+            LOGGER.error("Couldn't serialize object", e);
+            throw new SerializationException(e);
         }
-        send(bytes);
```

Comment:
Yeah, maybe I shouldn't have done that. I did it because Intellij complained in another file that I didn't use lambda's, so I added retrolambda to solve that warning, but then I got 2 new warnings in 2 other files that also should use retrolambda, so I fixed them. This was one of them. You can safely delete this code in the nfc pr